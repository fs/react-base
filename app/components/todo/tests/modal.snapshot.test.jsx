import React from 'react';
import renderer from 'react-test-renderer';
import TodoModal from '../modal';

jest.mock('components/modal', () => 'Modal');

describe('TodoModal', () => {
  let props;
  let component;
  const renderComponent = () => renderer.create(<TodoModal {...props} />);

  beforeEach(() => {
    props = {
      closeModal: () => { },
      createTodo: () => { },
      isOpen: true,
    };
  });

  it('renders correctly', () => {
    expect(renderComponent().toJSON()).toMatchSnapshot();
  });

  context('when form is invalid', () => {
    beforeEach(() => {
      component = renderComponent();

      component.getInstance().setState({ email: 'qwe' });
    });

    it('renders form with validation errors', () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
