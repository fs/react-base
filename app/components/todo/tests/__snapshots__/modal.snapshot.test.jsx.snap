// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoModal renders correctly 1`] = `
<Modal
  closeModal={[Function]}
  isOpen={true}
  title="New Task"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="taskName"
      >
        Task name
        : 
        
      </label>
      <input
        className="form-control"
        id="taskName"
        name="name"
        onChange={[Function]}
        placeholder="Task name..."
        type="text"
        value=""
      />
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Save
    </button>
  </form>
</Modal>
`;

exports[`TodoModal when form is invalid renders form with validation errors 1`] = `
<Modal
  closeModal={[Function]}
  isOpen={true}
  title="New Task"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="taskName"
      >
        Task name
        : 
        
      </label>
      <input
        className="form-control"
        id="taskName"
        name="name"
        onChange={[Function]}
        placeholder="Task name..."
        type="text"
        value=""
      />
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Save
    </button>
  </form>
</Modal>
`;
